lib/logstash/plugin_mixins/kafka/common.rb

```diff
@@ -42,6 +42,13 @@ def set_sasl_config(props)
       props.put("sasl.kerberos.service.name", sasl_kerberos_service_name) unless sasl_kerberos_service_name.nil?
       props.put("sasl.jaas.config", sasl_jaas_config) unless sasl_jaas_config.nil?
       props.put("sasl.client.callback.handler.class", sasl_client_callback_handler_class) unless sasl_client_callback_handler_class.nil?
+      props.put("sasl.oauthbearer.token.endpoint.url", sasl_oauthbearer_token_endpoint_url) unless sasl_oauthbearer_token_endpoint_url.nil?
+      props.put("sasl.oauthbearer.scope.claim.name", sasl_oauthbearer_scope_claim_name) unless sasl_oauthbearer_scope_claim_name.nil?
+      props.put("sasl.login.callback.handler.class", sasl_login_callback_handler_class) unless sasl_login_callback_handler_class.nil?
+      props.put("sasl.login.connect.timeout.ms", sasl_login_connect_timeout_ms.to_s) unless sasl_login_connect_timeout_ms.nil?
```

Comment:
It seems currently, `sasl.login.connect.timeout.ms`,  `sasl.login.read.timeout.ms` and `sasl.login.retry.backoff.*` apply only to OAUTHBEARER ([reference](https://kafka.apache.org/31/generated/admin_client_config.html#adminclientconfigs_sasl.login.retry.backoff.max.ms)). WDYT if we set them _only_ when `sasl.oauthbearer.*` params set?

Reply:
The references:
- https://kafka.apache.org/31/generated/producer_config.html
- https://kafka.apache.org/31/generated/consumer_config.html

Reply:
I couldn't find any evidence of these values being set actually harming anything. Given the language in the docs 
> The (optional) value in milliseconds for the external authentication provider connection timeout. Currently applies only to OAUTHBEARER.

the use of the work "currently" makes me think that they may extend it further at a later time. Trying to get in the middle here and add conditional logic does not seem worth the risk of *not* just letting the underlying client lib handle the values.

Reply:
Thanks for clarifying~!